module/zfs/zfs_ioctl.c

```diff
@@ -6962,12 +6965,17 @@ zfs_attach(void)
 	zfsdev_state_list->zs_minor = -1;
 
 	error = misc_register(&zfs_misc);
-	if (error != 0) {
-		printk(KERN_INFO "ZFS: misc_register() failed %d\n", error);
-		return (error);
+	if (error == -EBUSY) {
+		/*
+		 * Fallback to dynamic minor allocation in the event of a
+		 * collision with a reserved minor in linux/miscdevice.h.
+		 * In this case the kernel modules must be manually loaded.
+		 */
+		zfs_misc.minor = MISC_DYNAMIC_MINOR;
+		error = misc_register(&zfs_misc);
```

Comment:
keeping the existing printk (moved into a second if after the EBUSY one) and adding a second printk inside the if like this:
```
printk(KERN_INFO "ZFS: misc_register() with static minor %d failed %d, retrying with MISC_DYNAMIC_MINOR\n", ZFS_MINOR, error);
```

would make debugging issues here a lot easier I think (although they will probably trigger rarely if at all).